tests/Controller/HelperControllerTest.php

```diff
@@ -312,6 +316,67 @@ public function testsetObjectFieldValueAction()
         $this->assertEquals(200, $response->getStatusCode());
     }
 
+    public function testsetObjectFieldValueActionOnARelationField()
+    {
+        $object = new AdminControllerHelper_Foo();
+        $associationObject = new AdminControllerHelper_Bar();
+        $request = new Request([
+            'code' => 'sonata.post.admin',
+            'objectId' => 42,
+            'field' => 'bar',
+            'value' => 1,
+            'context' => 'list',
+        ], [], [], [], [], ['REQUEST_METHOD' => 'POST', 'HTTP_X_REQUESTED_WITH' => 'XMLHttpRequest']);
+
+        $fieldDescription = $this->prophesize(FieldDescriptionInterface::class);
+        $admin = $this->prophesize(AbstractAdmin::class);
+        $container = $this->prophesize(ContainerInterface::class);
+        $validator = $this->prophesize(ValidatorInterface::class);
+        $managerRegistry = $this->prophesize(ManagerRegistry::class);
+        $objectManager = $this->prophesize(ObjectManager::class);
+        $classMetadata = $this->prophesize(ClassMetadata::class);
+        $adminExtension = $this->prophesize(SonataAdminExtension::class);
+        $twig = $this->prophesize(\Twig_Environment::class);
+        $pool = $this->prophesize(Pool::class);
+        $propertyAccessor = $this->prophesize(PropertyAccessor::class);
```

Comment:
Consider mocking less things that you do not own.

Reply:
Okay, will try. But there are some like container or twig, that could be hard not to mock them

Reply:
Yeah, please don't try too hard, just do what you can do easily (not sure there is any but… you have the code).